### PR TITLE
imported Truth.assertThat

### LINKED ISSUE #766: Please import `Truth.assertThat` directly:
Please import `Truth.assertThat` directly:

```kt
import com.google.common.truth.Truth.assertThat
```

```kt
assertThat(engineOne).isSameInstanceAs(engineTwo)
```

_Originally posted by @kevinmost in https://github.com/google/android-fhir/pull/749#discussion_r699401052_

### PR DESCRIPTION
**Pull request by Harsh9149**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #766 

**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/barcode/QuestionnaireItemBarCodeReaderViewHolderFactoryInstrumentedTest.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ import com.google.android.fhir.datacapture.R
 import com.google.android.fhir.datacapture.views.QuestionnaireItemViewHolder
 import com.google.android.fhir.datacapture.views.QuestionnaireItemViewItem
 import com.google.common.truth.Truth
```
Comment (epicadk) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/barcode/QuestionnaireItemBarCodeReaderViewHolderFactoryInstrumentedTest.kt:29]: Do we still require this import?

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnairesTest.kt [hunk 2]
```diff
@@ -17,6 +17,7 @@
 package com.google.android.fhir.datacapture
 
 import com.google.common.truth.Truth
```
Comment (epicadk) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnairesTest.kt:19]: ditto

File: engine/src/test-common/java/com/google/android/fhir/resource/TestingUtils.kt [hunk 3]
```diff
@@ -25,6 +25,7 @@ import com.google.android.fhir.db.impl.dao.SquashedLocalChange
 import com.google.android.fhir.search.Search
 import com.google.android.fhir.sync.DataSource
 import com.google.common.truth.Truth
```
Comment (epicadk) [inline engine/src/test-common/java/com/google/android/fhir/resource/TestingUtils.kt:27]: same

File: engine/src/test/java/com/google/android/fhir/FhirEngineProviderTest.kt [hunk 4]
```diff
@@ -19,6 +19,7 @@ package com.google.android.fhir
 import androidx.test.core.app.ApplicationProvider
 import androidx.test.platform.app.InstrumentationRegistry
 import com.google.common.truth.Truth
```
Comment (epicadk) [inline engine/src/test/java/com/google/android/fhir/FhirEngineProviderTest.kt:21]: ditto

**Review by epicadk — COMMENTED** [review]
a few comments

**Review by jingtang10 — CHANGES_REQUESTED** [review]
thanks for this - please address @epicadk's comments!

thansk @epicadk for the review.

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/barcode/QuestionnaireItemBarCodeReaderViewHolderFactoryInstrumentedTest.kt (hunk 1 above)
Comment (Harsh9149) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/barcode/QuestionnaireItemBarCodeReaderViewHolderFactoryInstrumentedTest.kt]: @epicadk I have run the spotless and the imports are gone. PTAL.

**Review by jingtang10 — APPROVED** [review]
@Harsh9149 thanks for the PR 👍

**Comment by jingtang10** [comment]
please tick the remaining items in the description:

 I have run ./gradlew check and ./gradlew connectedCheck to test my changes locally
 I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

**Comment by Harsh9149** [comment]
Hi @jingtang10 is android fhir is participating in upcoming Gsoc'22.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnairesTest.kt (hunk 2 above)
Comment (Harsh9149) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnairesTest.kt]: Done

File: engine/src/test-common/java/com/google/android/fhir/resource/TestingUtils.kt (hunk 3 above)
Comment (Harsh9149) [reply engine/src/test-common/java/com/google/android/fhir/resource/TestingUtils.kt]: Done

File: engine/src/test/java/com/google/android/fhir/FhirEngineProviderTest.kt (hunk 4 above)
Comment (Harsh9149) [reply engine/src/test/java/com/google/android/fhir/FhirEngineProviderTest.kt]: Done

**Comment by Harsh9149** [comment]
> Hi @jingtang10 is android fhir is participating in upcoming Gsoc'22.

@f-odhiambo

**Comment by Harsh9149** [comment]
@epicadk I am not getting what is going wrong in my PR. My build test is failing can you please help me in resolving the error.

**Comment by jingtang10** [comment]
> @epicadk I am not getting what is going wrong in my PR. My build test is failing can you please help me in resolving the error.

The instrumentation tests have been timing out across different PRs. Not sure if this is a transient issue of github.

**Comment by Harsh9149** [comment]
> > @epicadk I am not getting what is going wrong in my PR. My build test is failing can you please help me in resolving the error.
> 
> The instrumentation tests have been timing out across different PRs. Not sure if this is a transient issue of github.

@jingtang10 is there a way to resolve this issue please let me know and also can you rerun the CI check again it might resolve the issue.

**Comment by epicadk** [comment]
> > > @epicadk I am not getting what is going wrong in my PR. My build test is failing can you please help me in resolving the error.
> > 
> > The instrumentation tests have been timing out across different PRs. Not sure if this is a transient issue of github.
> 
> @jingtang10 is there a way to resolve this issue please let me know and also can you rerun the CI check again it might resolve the issue.

Already tried this.

**Comment by Harsh9149** [comment]
@jingtang10 please approve the CI check

**Comment by jingtang10** [comment]
@Harsh9149 as you can see in #1120 we are experiencing some issues with some of our tests. so while this pr is approved it probably won't be merged until that issue is resolved - hopefully soon.

**Pull request merged**